lib/client/keystore.go

```diff
@@ -559,6 +569,33 @@ func (ms *MemKeyStore) AddKey(key *Key) error {
 
 	ms.keys[key.ProxyHost][key.Username][key.ClusterName] = keyCopy
 
+	// If virtual profile env vars are set, we also write the key components to the FS
+	var envName string
+
```

Comment:
What would be the read behaviour?

- read in mem, then the virtual profile if it's a miss
- read the virtual profile, then in mem if it's a miss
- only in the virtual profile?

I'm worried about how backward compatible this would be.

Reply:
Also what's the best behaviour when tryinig to delete a missing key? (e.g. the key is in-memory but not on the FS)

Reply:
I don't think we can implement proper read/delete because reading/deleting requires looking up the env var, which requires knowing the kube,app,db name, which requires a read operation.

Reply:
For delete we could just fetch and delete all env vars prefixed with `TSH_VIRTUAL_PATH`, but for read I'm worried we cannot get back the original name reliably e.g. an app name with a `-` in it.

Reply:
I didn't realize that's how `TSH_VIRTUAL_PATH`s were working...I don't think there is a good way to make this work then :/